src/Microsoft.Health.Api/Features/Audit/AuditEventTypeMapping.cs

```diff
@@ -74,9 +74,20 @@ public Task StartAsync(CancellationToken cancellationToken)
                     return (ad.ControllerName, ad.ActionName, Attribute: attribute);
                 })
                 .Where(item => item.Attribute != null)
+                .Distinct()
+                .GroupBy(item => (item.ControllerName, item.ActionName))
+                .Select(item =>
+                {
+                    if (item.Count() != 1)
+                    {
+                        throw new DuplicateActionForAuditEventException(item.Key.ControllerName, item.Key.ActionName);
```

Comment:
I appreciate the unique exception. You are also welcome to re-use exceptions where appropriate like perhaps `InvalidOperationException`. I mention this because (1) having too many very specific exceptions may be overwhelming to users and (2) creating exceptions can be kind of a pain as they should expose a [few standard ctors](https://docs.microsoft.com/en-us/dotnet/standard/exceptions/how-to-create-user-defined-exceptions), or even make them serializable as a best practice.

Reply:
Ah makes sense. I've kept in this exception since it's already there, but updated it to have the correct constructors